src/test_workflow/integ_test/topology.py

```diff
@@ -0,0 +1,79 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Any, Generator, List
+
+from manifests.bundle_manifest import BundleManifest
+from test_workflow.dependency_installer import DependencyInstaller
+from test_workflow.integ_test.local_test_cluster import LocalTestCluster
+from test_workflow.test_recorder.test_recorder import TestRecorder
+
+
+class Topology:
+
+    def __init__(
+        self,
+        number_of_cluster: int,
+        dependency_installer: DependencyInstaller,
+        work_dir: Path,
+        component_name: str,
+        additional_cluster_config: dict,
+        bundle_manifest: BundleManifest,
+        security_enabled: bool,
+        component_test_config: str,
+        test_recorder: TestRecorder
+    ) -> None:
```

Comment:
Most of these parameters aren't needed for a topology to exist, but are only needed for executing tests, so it should be broken up. 

With the refactor suggested above by @kotwanikunal you will be getting a `InputManifest.TestTopology` from `manifest....topology` with a bunch of information such as the number of nodes. So this class should only take that as input. Another method `create` would start the cluster and manage its shutdown. And another method will execute tests, taking as input whatever it needs to execute.